### DISCUSSION TITLE
Using useClickOutside with Select

### DISCUSSION THREAD

**Discussion by ShakedJuran** [mantinedev/mantine#4187] [Q&A]
When I use useClickOutside with a Select component it seems that even when I click the scroll bar or any of the items in the dropdown will trigger the click outside callback.
Am I missing something, is there a way to do this?

Edit: Atm I wrap the Select component with a `<div `ref={ref}/>` using useClickOutside on it, is this the only way to do this?

**Comment by mstone121** [comment]
I'm experiencing the same thing. My guess is that it's caused by the `Select` `Popover` being rendered in a portal which is outside the element with `useClickOutside`.

According to the documentation, it's possible to stop elements from triggering an outside click by passing them as the third parameter to `useClickOutside`. See [use-click-outside/#multiple-nodes](https://mantine.dev/hooks/use-click-outside/#multiple-nodes). In my case, the `Select` components are nested deeply in the component that uses `useClickOutside` and there are lots of them, so this isn't really feasible. 

In the source of `useClickOutside`, it seems that if any array is passed as a third argument, then elements with the `data-ignore-outside-clicks` attribute are also ignored.
https://github.com/mantinedev/mantine/blob/7f66edc5887958fd5cb99a93843483e4154a0133/src/mantine-hooks/src/use-click-outside/use-click-outside.ts#L17
And `SelectItems` used to have this attribute until this commit: https://github.com/mantinedev/mantine/commit/8028fefd6ab1d01b83ac272045954acafb90841e when it was commented out. I wonder if @rtivital could shed some light on why this was done.
